internal/config/config_test.go

```diff
@@ -33,3 +34,20 @@ func TestConfig_Load(t *testing.T) {
 	assert.NotEmpty(t, cfg.GetString("username"))
 	assert.NotEmpty(t, cfg.GetString("password"))
 }
+
+func TestConfig_LoadKeyring(t *testing.T) {
+	cfg := New()
+	tmpFile, err := os.CreateTemp(os.TempDir(), "")
+	assert.NoError(t, err)
+	_, err = tmpFile.WriteString("username: sdkfo")
+	assert.NoError(t, err)
+
+	err = keyring.Set("upctl", "sdkfo", "foo")
```

Comment:
This will end up in the systems default keyring, so could use more descriptive username, e.g `unittest`, and a cleanup function for removing this with [t.Cleanup](https://pkg.go.dev/testing@master#T.Cleanup).

Reply:
will do. I will update the docker image and test it before pushing. Keep you posted.

Reply:
docker image updated: took a sample from [go-keyring](https://github.com/zalando/go-keyring/blob/master/.github/workflows/go.yml)
cleanup implemented: clean only the test user that was set